lib/rust/lexer/definition/src/token.rs

```diff
@@ -162,6 +246,28 @@ impl Token {
         let shape  = Shape::Unrecognized(str);
         Token{shape,length,offset}
     }
+
+    /// Construct a token representing a disable comment.
+    pub fn DisableComment(text:impl Str, offset:usize) -> Token {
+        let str    = text.into();
+        let length = str.chars().count() + 1;
+        let shape  = Shape::DisableComment(str);
+        Token{shape,length,offset}
+    }
+
+    /// Construct a token representing a documentation comment.
+    pub fn DocComment(lines:Vec<Token>, indent:usize, offset:usize) -> Token {
+        let length = lines.iter().map(|line| {
+            match line.shape {
+                Shape::Line{..}     => indent + line.source_length(),
+                Shape::BlankLine(_) => line.source_length(),
+                _                   =>
+                    unreachable_panic!("Tokens in a doc comment should always be lines."),
```

Comment:
tbh in such case I'd sacrifice the alignment and make it 1-liner. What do you think ?